ProcessLib/LiquidFlow/LiquidFlowLocalAssembler.h

```diff
@@ -81,8 +91,10 @@ class LiquidFlowLocalAssembler : public LiquidFlowLocalAssemblerInterface
         int const gravitational_axis_id,
         double const gravitational_acceleration,
         double const reference_temperature,
-        LiquidFlowMaterialProperties const& material_propertries)
-        : _element(element),
+        LiquidFlowMaterialProperties const& material_propertries,
+        StaggeredCouplingTerm* coupling_term)
```

Comment:
It's unfortunate that now we need to store the coupling term in the local assemblers.
I believe that one can circumvent this; it's not hard, but certainly not trivial.
@wenqing if you want to change this now and if you see the solution without much thought, please make the change. Otherwise, could you please add a `// TODO remove additional member` to the source code? Then I can change it in the future.

Reply:
Yes a pointer type is added to all assembler instances. 
I prefer to move the integration data out of the local assembler and make the  local assembler unique for each process. By this way, the repeated computation and memory allocation for  integration data among different process, which use the same element order, can be avoid and consequently this coulping term pointer is only assigned once for each process.

I added a TODO comment  before the declaration of _coupling_term with the meanings of the above words.

Reply:
Agree to the fact that coupling term being in local assemblers. We shall see how it develops.